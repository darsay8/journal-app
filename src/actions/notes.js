import { db } from '../firebase/firebaseConfig';
import { collection, addDoc } from 'firebase/firestore';
import { types } from '../types/types';

export const startNewNote = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    const newNote = {
      title: '',
      body: '',
      date: new Date().getTime(),
    };

    try {
      const docRef = await addDoc(collection(db, `${uid}/journal/notes`), newNote);

      dispatch(activeNote(docRef.id, newNote));
    } catch (e) {
      console.log(e);
    }
  };
};

export const activeNote = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
});

export const setNotes = notes => ({
  type: types.notesLoad,
  payload: notes,
});
