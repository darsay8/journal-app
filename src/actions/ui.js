import { types } from '../types/types';

export const setError = err => ({
  type: types.uiSetError,
  payload: {
    msgError: err,
  },
});

export const removeError = () => ({
  type: types.uiRemoveError,
});
