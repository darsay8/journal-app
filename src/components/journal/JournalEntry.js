const JournalEntry = () => {
  const imgUrl =
    'https://geniusbeauty.com/wp-content/uploads/2008/11/california-wallpaper-nature-10.jpg';

  return (
    <div className="journal__entry animate__animated animate__fadeIn animate__faster">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: 'cover',
          backgroundImage: `url(${imgUrl})`,
        }}
      ></div>

      <div className="journal__entry-body">
        <p className="journal__entry-title">Entry</p>
        <p className="journal__entry-content">
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
        </p>
      </div>
      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>
    </div>
  );
};

export default JournalEntry;
