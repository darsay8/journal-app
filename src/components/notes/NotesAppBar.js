import React from 'react';

const NotesAppBar = () => {
  return (
    <div className="notes__appbar">
      <span>Feb 28th, 2022</span>
      <input id="fileSelector" name="file" type="file" style={{ display: 'none' }} />
      <div>
        <button className="btn">Picture</button>
        <button className="btn">Save</button>
      </div>
    </div>
  );
};

export default NotesAppBar;
