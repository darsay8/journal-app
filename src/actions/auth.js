import {
  getAuth,
  signInWithPopup,
  signInWithEmailAndPassword,
  createUserWithEmailAndPassword,
  updateProfile,
  signOut,
} from '@firebase/auth';
import Swal from 'sweetalert2';
import { googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from '../types/types';
import { notesLogout } from './notes';
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = (email, password) => {
  return dispatch => {
    dispatch(startLoading());
    const auth = getAuth();
    signInWithEmailAndPassword(auth, email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch(e => {
        dispatch(finishLoading());
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return dispatch => {
    dispatch(startLoading());
    const auth = getAuth();
    createUserWithEmailAndPassword(auth, email, password)
      .then(async ({ user }) => {
        await updateProfile(user.auth.currentUser, { displayName: name });
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch(e => {
        dispatch(finishLoading());
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return dispatch => {
    const auth = getAuth();
    signInWithPopup(auth, googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch(e => {
        console.log('Error: ', e);
      });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: { uid, displayName },
});

export const startLogOut = () => {
  return async dispatch => {
    const auth = getAuth();
    await signOut(auth);
    dispatch(logout());
    dispatch(notesLogout());
  };
};

export const logout = () => ({ type: types.logout });
