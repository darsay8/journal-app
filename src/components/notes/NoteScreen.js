import { useEffect, useRef } from 'react';
import { useSelector } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import NotesAppBar from './NotesAppBar';

const NoteScreen = () => {
  const { active: note } = useSelector(state => state.notes);

  const [formValues, handleInputChange, reset] = useForm(note);
  const { body, title } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <form action="">
          <input
            type="text"
            placeholder="Some awesome title"
            className="notes__title-input"
            autoComplete="off"
            name="title"
            value={title}
            onChange={handleInputChange}
          />
          <textarea
            placeholder="What happened today"
            className="notes__textarea"
            name="body"
            value={body}
            onChange={handleInputChange}
          ></textarea>
        </form>

        {note.url && (
          <div className="notes__image">
            <img src={note.url} alt="landscape" />
          </div>
        )}
      </div>
      <button className="btn btn-danger">Delete</button>
    </div>
  );
};

export default NoteScreen;
