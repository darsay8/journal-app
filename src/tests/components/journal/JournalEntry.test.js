import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import 'jsdom-global/register';

import JournalEntry from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
  id: 10,
  date: 0,
  title: 'Title',
  body: 'Body Text',
  url: 'https://testimages.com/image.jpg',
};

const wrapper = mount(
  <Provider store={store}>
    <JournalEntry note={note} />
  </Provider>,
);

describe('<JournalEntry/> Test', () => {
  test('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should active note ', () => {
    // wrapper.find('.journal__entry').simulate('click');
    wrapper.find('.journal__entry').prop('onClick')();

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, { ...note }));
  });
});
