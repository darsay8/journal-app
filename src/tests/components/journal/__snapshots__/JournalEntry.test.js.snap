// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalEntry/> Test should match with snapshot 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    />
    <p
      className="journal__entry-content"
    />
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Friday
    </span>
    <h4>
      25th
    </h4>
  </div>
</div>
`;
